### PR TITLE
feat: Update chart to support upstream images

### PR DESCRIPTION
**Pull request by beeradb**
This PR updates the sourcify chart with changes to support using upstream images. The changes are as follows:

- add servers.yaml, sourcify-chains.json, and local.js as Configmaps and mount them to the server container.
- Rework container image structure in both values.yaml and the corresponding deployments.
- Some minor linting of trailing spaces & newlines.

### CODE REVIEW

File: charts/values.yaml [hunk 1]
```diff
@@ -86,7 +86,37 @@ serverRepository:
...
+          "sourcifyName": "Hedera Previewnet",
+          "supported": true
+        },
+        "298": {
+          "sourcifyName": "Hedera Localnet",
+          "supported": true,
+          "rpc": [
+            "http://host.docker.internal:7546"
+          ]
+        }
```
Comment (acuarica) [inline charts/values.yaml:118]: For integration and production this network should be removed.

File: charts/templates/server-repository/configmap.yaml [hunk 2]
```diff
@@ -35,4 +35,20 @@ metadata:
...
+  local.js: |
+    module.exports = {
+      corsAllowedOrigins: [
+          /^https?:\/\/(?:.+\.)?sourcify.dev$/, // sourcify.dev and subdomains
+          /^https?:\/\/(?:.+\.)?sourcify.eth$/, // sourcify.eth and subdomains
+          /^https?:\/\/(?:.+\.)?sourcify.eth.link$/, // sourcify.eth.link and subdomains
+          /^https?:\/\/(?:.+\.)?ipfs.dweb.link$/, // dweb links used by Brave browser etc.
+          /^https?:\/\/(?:.+\.)?hedera-devops.com$/, // swirldslabs-devops
+          /^https?:\/\/(?:.+\.)?hashscan.io$/, // hashscan
+          process.env.NODE_ENV !== "production" && /^https?:\/\/localhost(?::\d+)?$/, // localhost on any port /^{{ .Values.serverRepository.config.SERVER_URL }}$/
```
Comment (acuarica) [inline charts/templates/server-repository/configmap.yaml:52]: also this one

File: charts/templates/server-repository/configmap.yaml [hunk 3]
```diff
@@ -20,7 +20,7 @@ data:
   TESTING: {{ .Values.serverRepository.config.TESTING | quote }}
   NODE_ENV: {{ .Values.serverRepository.config.NODE_ENV | default "development" }}
   UI_DOMAIN_NAME: {{ .Values.serverRepository.config.UI_DOMAIN_NAME }}
```
Comment (acuarica) [inline charts/templates/server-repository/configmap.yaml:22]: Can we delete `TESTING`, `UI_DOMAIN_NAME` and `REPOSITORY_PATH`?

Can we change the default of `NODE_ENV` to `production`?

**Review by acuarica — CHANGES_REQUESTED** [review]
lgtm thanks for taking care of this.

I left some improvements I didn't mention before.

File: charts/templates/server-repository/deployment.yaml [hunk 4]
```diff
@@ -34,12 +34,12 @@ spec:
...
         {{- toYaml .Values.serverRepository.podSecurityContext | nindent 8 }}
-      {{- end }}  
-      containers:        
+      {{- end }}
+      containers:
         - name: {{ printf "%s-%s" .Chart.Name "server" }}
           securityContext:
             {{- toYaml .Values.serverRepository.securityContext | nindent 12 }}
-          image: "{{ .Values.image.repository }}:{{ .Values.serverRepository.image.server.tag | default "server-latest" }}"
+          image: "{{ .Values.serverRepository.image.server.repository }}:{{ .Values.serverRepository.image.server.tag | default "server-latest" }}"
```
Comment (acuarica) [inline charts/templates/server-repository/deployment.yaml:42]: Same with `ui` image name.

File: charts/templates/server-repository/configmap.yaml [hunk 5]
```diff
@@ -8,19 +8,16 @@ metadata:
...
-  {{- end }}    
+  {{- end }}
 data:
-  REPOSITORY_PATH: {{ .Values.serverRepository.config.REPOSITORY_PATH }}
   SERVER_URL: {{ .Values.serverRepository.config.SERVER_URL }}
   REPOSITORY_SERVER_URL: {{ .Values.serverRepository.config.REPOSITORY_SERVER_URL }}
   SERVER_PORT: {{ .Values.serverRepository.service.server.port | quote }}
   SERVER_EXTERNAL_PORT: {{ .Values.serverRepository.service.server.port | quote }}
   REPOSITORY_SERVER_PORT: {{ .Values.serverRepository.service.repository.port | quote }}
   REPOSITORY_SERVER_EXTERNAL_PORT: {{ .Values.serverRepository.service.repository.port | quote }}
```
Comment (acuarica) [inline charts/templates/server-repository/configmap.yaml:18]: These `*_PORT` config should also be removed, they are not used by any application.

**Review by acuarica — DISMISSED** [review]
lgtm

**Review by acuarica — APPROVED** [review]
lgtm

**Review by jeromy-cannon — APPROVED** [review]

**Pull request merged**